cli/args/flags.rs

```diff
@@ -1976,6 +1976,13 @@ Directory arguments are expanded to all contained files matching the glob
         .value_hint(ValueHint::FilePath)
         .help("Write a JUnit XML test report to PATH. Use '-' to write to stdout which is the default when PATH is not provided.")
     )
+    .arg(
+      Arg::new("junit-step-mode")
+      .long("junit-step-mode")
+      .default_value("property")
+      .help("Controls how steps are represented in the JUnit report")
+      .value_parser(["property", "testcase", "testcase-ignore-parent-results", "testcase-exclude-parents"])
+    )
```

Comment:
I'm not sure I'm a fan of adding another flag that controls how only Junit reporter treats test steps failures. Maybe we can decide on one that makes the most sense? Test steps were meant to act like `it` functions from BDD libraries.

Reply:
I think it's important we give flexibility because there's nothing that enforces that expectation. In mocha failures in `describe` blocks cause the entire test run to bail early preventing people from including testing logic in them. `Deno.test` has no such limitation, and so assertions can be included at any layer of the tree.  

If someone were to include assertions in a parent test, `testcase-ignore-parent-results` and `testcase-exclude-parents` would both be inappropriate because failures in those assertions would no longer be reported. `testcase` has the flaw of messing up the failure counts for those who do treat steps as a BDD-style tool because a failure in the step would propagate up to the parent and count as two failures when there was only one.

Essentially, `Deno.test` is extremely (perhaps overly) flexible in it's usage, and so all we can do is expose the flexibility to the user.